src/AppInstallerCLICore/Workflows/UninstallFlow.cpp

```diff
@@ -201,14 +203,28 @@ namespace AppInstaller::CLI::Workflow
             catch (const wil::ResultException& re)
             {
                 context.Add<Execution::Data::OperationReturnCode>(re.GetErrorCode());
-                context.Reporter.Error() << Resource::String::UninstallFailedWithCode << ' ' << re.GetErrorCode() << std::endl;
-                AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_EXEC_UNINSTALL_COMMAND_FAILED);
+                context << ReportUninstallerResult("MSIXUninstall"sv, re.GetErrorCode(), /* isHResult */ true);
+                return;
             }
         }
 
         context.Reporter.Info() << Resource::String::UninstallFlowUninstallSuccess << std::endl;
     }
 
+    void ShellExecuteUninstall(Execution::Context& context)
+    {
+        context <<
+            Workflow::ShellExecuteUninstallImpl <<
+            ReportUninstallerResult("UninstallString", APPINSTALLER_CLI_ERROR_EXEC_UNINSTALL_COMMAND_FAILED);
```

Comment:
nit: I don't see the value in splitting this for two types when MSIX call its `ReportUninstallerResult` in `ExecuteUninstaller`.  Why not just put all of them in `ExecuteUninstaller`?

Reply:
Changed so that it directly calls from ExecuteUninstaller